pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -108,49 +110,92 @@ func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	logger = logger.WithValues("deployment", dep.Name)
 
-	ing := &netv1.Ingress{}
-	ing.Name = util.FindOwnerKind(spc.OwnerReferences, "Ingress")
-	ing.Namespace = req.Namespace
-	logger = logger.WithValues("ingress", ing.Name)
-	if ing.Name != "" {
-		logger.Info("getting owner ingress")
-		if err = p.client.Get(ctx, client.ObjectKeyFromObject(ing), ing); err != nil {
-			return result, client.IgnoreNotFound(err)
+	if strings.HasPrefix(spc.Name, NginxNamePrefix) {
+		nic := &v1alpha1.NginxIngressController{}
+		nic.Name = util.FindOwnerKind(spc.OwnerReferences, "NginxIngressController")
+
+		if nic.Name != "" {
+			logger.Info("getting owner nginx ingress controller")
+			if err = p.client.Get(ctx, client.ObjectKeyFromObject(nic), nic); err != nil {
+				return result, client.IgnoreNotFound(err)
+			}
 		}
-	}
 
-	managed, err := p.ingressManager.IsManaging(ing)
-	if err != nil {
-		return result, fmt.Errorf("determining if ingress is managed: %w", err)
-	}
+		if nic.Name == "" || nic.Spec.IngressClassName == "" || nic.Spec.DefaultSSLCertificate.KeyVaultURI == "" {
+			logger.Info("cleaning unused placeholder pod deployment")
+			logger.Info("getting placeholder deployment")
+			toCleanDeployment := &appsv1.Deployment{}
+			if err = p.client.Get(ctx, client.ObjectKeyFromObject(dep), toCleanDeployment); err != nil {
+				return result, client.IgnoreNotFound(err)
+			}
+			if manifests.HasTopLevelLabels(toCleanDeployment.Labels) {
+				logger.Info("deleting placeholder deployment")
+				err = p.client.Delete(ctx, toCleanDeployment)
+				return result, client.IgnoreNotFound(err)
+			}
+		}
+
+		// Manage a deployment resource
+		logger.Info("reconciling placeholder deployment for secret provider class")
+		if err = p.buildDeployment(ctx, dep, spc, nic.Name); err != nil {
+			err = fmt.Errorf("building deployment: %w", err)
+			p.events.Eventf(nic, "Warning", "FailedUpdateOrCreatePlaceholderPodDeployment", "error while building placeholder pod Deployment needed to pull Keyvault reference: %s", err.Error())
+			logger.Error(err, "failed to build placeholder deployment")
+			return result, err
+		}
+
+		if err = util.Upsert(ctx, p.client, dep); err != nil {
+			p.events.Eventf(nic, "Warning", "FailedUpdateOrCreatePlaceholderPodDeployment", "error while creating or updating placeholder pod Deployment needed to pull Keyvault reference: %s", err.Error())
+			return result, err
+		}
+	} else {
```

Comment:
pretty much all this branching logic is exactly the same except one is for pushing events to an ingress resource and the other is to nic resources. Instead of having an if else here we should have a function that gets the responsible object then just performs the exact same logic for both. 

nic and ingress are both client.Object interfaces so you can treat them the same